### DISCUSSION TITLE
Timer / stopwatch in presenter view

### DISCUSSION THREAD

**Discussion by Dimension4** [marp-team/marp#65] [Ideas] [answered]
Currently we only have the local time in presenter view. It would be really helpful to have a timer with start, stop and reset functionality so we can see the elapsed time. Is this possible somehow?

**Comment by yhatt** [comment]
First [I had planned to toggle the clock and timer by click](https://github.com/marp-team/marp-cli/pull/157#issuecomment-571014937), but I preferred releasing the incompleted presenter view including only the local time, [based on the feedback by user who wanted the presenter view](https://github.com/marp-team/marp-cli/pull/157#issuecomment-571107901).

> Also as engineer I understand that after initial release it is possible to create ticket of improvement of existing solution (actually several independent improvements: add timer, add support for templates and so on).
>
> _-- @asm0dey said in https://github.com/marp-team/marp-cli/pull/157#issuecomment-571107901_

Welcome to create a request issue (and contribution) into [Marp CLI issue tracker](https://github.com/marp-team/marp-cli/issues/)!

**Reply by Dimension4** [reply]
I'll give it a try, although I've no experience in js/html so you might have to work harder on the PR 😉

**Comment by asm0dey** [comment]
@yhatt and I'm grateful for your decision! I used it countless number of times and even have a template repo for my presentations already!

**Comment by yhatt** [comment, marked as answer]
Thanks to awesome contributions by @chrisns, [Marp CLI v1.7.0](https://github.com/marp-team/marp-cli/releases/tag/v1.7.0) is getting a timer in the presenter view of HTML output (bespoke template)!

The timer is in opposite side of the current time view. It can reset to zero by click.

![](https://user-images.githubusercontent.com/3993388/155341980-d5acb427-2ba9-4ed5-991c-432a1d559333.png)
